Source/Kernel/Concepts/Observation/FailedPartitions.cs

```diff
@@ -12,46 +13,49 @@ namespace Cratis.Chronicle.Concepts.Observation;
 public class FailedPartitions
 {
     readonly List<FailedPartition> _resolvedPartitions = [];
-    List<FailedPartition> _partitions = [];
+    readonly Dictionary<Key, FailedPartition> _partitions = [];
 
     /// <summary>
     /// Gets or sets the failed partitions for the observer.
     /// </summary>
     public IEnumerable<FailedPartition> Partitions
     {
-        get => _partitions;
-        set => _partitions = new(value);
+        get => _partitions.Values;
+        set
+        {
+            _partitions.Clear();
+            foreach (var failedPartition in value)
+            {
+                var key = failedPartition.Partition;
+                _partitions.Add(key, failedPartition);
+            }
```

Comment:
This could be with LINQ:

```suggestion
            _partitions = failedPartitions.ToDictionary(_ => _.Partition, _ => _);
```